### ISSUE TITLE
allow verifying sub-module

### ISSUE THREAD

**Issue by bukzor-sentryio** [microsoft/pyright#4999]
**Is your feature request related to a problem? Please describe.**

I've requested pyright to verify a submodule of my larger module, and it gets confused.

First showing `--verifytypes` working correctly in its intended use case:

```
$ ls -la hugelib
total 8
drwxr-xr-x 5 buck staff 160 Apr 24 10:57 .
drwxr-xr-x 4 buck staff 128 Apr 24 10:57 ..
-rw-r--r-- 1 buck staff   0 Apr 24 10:57 py.typed
-rw-r--r-- 1 buck staff  11 Apr 24 10:57 x.py
-rw-r--r-- 1 buck staff  17 Apr 24 10:57 y.py

$ tail hugelib/x.py hugelib/y.py
==> hugelib/x.py <==
x: int = 3

==> hugelib/y.py <==
y: str = "wai"

$ PYTHONPATH=$PWD pyright --verifytypes hugelib
Module name: "hugelib"
Package directory: "$PWD/hugelib"
Module directory: "$PWD/hugelib"
Path of py.typed file: "$PWD/hugelib/py.typed"

Public modules: 2
   hugelib.x
   hugelib.y

Symbols used in public interface:

Symbols exported by "hugelib": 2
  With known type: 2
  With ambiguous type: 0
  With unknown type: 0
  Functions without docstring: 0
  Functions without default param: 0
  Classes without docstring: 0

Other symbols referenced but not exported by "hugelib": 0
  With known type: 0
  With ambiguous type: 0
  With unknown type: 0
```

If I instead request it to verify a submodule of `hugelib` it succeeds in resolving the package, but gets confused when resolving modules:

```
$ PYTHONPATH=$PWD pyright --verifytypes hugelib.y
Module name: "hugelib.y"
Package directory: "$PWD/hugelib"
Module directory: "$PWD/hugelib"
Path of py.typed file: "$PWD/hugelib/py.typed"
   error: Could not resolve module "hugelib.y.x"
   error: Could not resolve module "hugelib.y.y"

Symbols used in public interface:

Symbols exported by "hugelib.y": 0
  With known type: 0
  With ambiguous type: 0
  With unknown type: 0
  Functions without docstring: 0
  Functions without default param: 0
  Classes without docstring: 0

Other symbols referenced but not exported by "hugelib.y": 0
  With known type: 0
  With ambiguous type: 0
  With unknown type: 0

Type completeness score: 0%
```

**Describe the solution you'd like**
Ideally the argument to `--verifytypes` would behave analogously to `python -m`, restricting its scan and report to a submodule when a submodule is specified. Hypothetically:

```
$ PYTHONPATH=$PWD pyright --verifytypes hugelib.y

Module name: "hugelib"
Package directory: "$PWD/hugelib"
Module file: "$PWD/hugelib/y.py"
Path of py.typed file: "$PWD/hugelib/py.typed"

Public modules: 1
   hugelib.y

Symbols used in public interface:

Symbols exported by "hugelib.y": 1
  With known type: 1
  With ambiguous type: 0
  With unknown type: 0
  Functions without docstring: 0
  Functions without default param: 0
  Classes without docstring: 0

Other symbols referenced but not exported by "hugelib.y": 0
  With known type: 0
  With ambiguous type: 0
  With unknown type: 0
```

**Additional context**

This will be convenient in real-world project where the several modules under a package have various typing maturity. For now I can work around this with `grep -v` :)

**Comment by erictraut** [comment]
I'm unable to repro a problem here. I've tried with several common libraries with submodules, and it seems to work fine.

Is your library publicly available? I'll need repro steps if you would like me to investigate further.

You may be able to gather additional clues about what's going wrong by using the `--verbose` command-line option.

**Comment by erictraut** [comment]
I reread your bug report and figured out why I wasn't able to repro the problem. I can now repro it, so I'll dig into it further.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.306, which I just published.
